pkg/controller/v1beta1/inferenceservice/reconcilers/keda/keda_reconciler.go

```diff
@@ -0,0 +1,212 @@
+	}
+	// override metric configuration from componentExtension if it is set
+	if componentExt.ScaleMetric != nil {
+		triggerType = string(*componentExt.ScaleMetric)
+	}
+	if componentExt.ScaleMetricType != nil {
+		metricType = *componentExt.ScaleMetricType
+	}
+	if componentExt.ScaleTarget != nil {
+		scaleTarget = *componentExt.ScaleTarget
+	}
+	// override metric configuration from componentExtension.ScalerSpec if it is set
+	if componentExt.ScalerSpec != nil {
+		if componentExt.ScalerSpec.ScaleMetric != nil {
+			triggerType = string(*componentExt.ScalerSpec.ScaleMetric)
+		}
+		if componentExt.ScalerSpec.ServerAddress != "" {
+			serverAddress = componentExt.ScalerSpec.ServerAddress
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			scaleTarget = *componentExt.ScalerSpec.ScaleTarget
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			metricType = *componentExt.ScalerSpec.ScaleMetricType
+		}
+	}
+
+	trigger := kedav1alpha1.ScaleTriggers{
+		Type:     triggerType,
+		Metadata: map[string]string{},
+	}
+
+	// set trigger metadata for prometheus and graphite triggers
+	if triggerType == "prometheus" || triggerType == "graphite" {
+		if serverAddress != "" {
+			trigger.Metadata["serverAddress"] = serverAddress
+		}
+		if componentExt.ScalerSpec != nil {
+			if componentExt.ScalerSpec.MetricQuery != "" {
+				trigger.Metadata["query"] = componentExt.ScalerSpec.MetricQuery
+			}
+			if componentExt.ScalerSpec.QueryParameters != "" {
+				trigger.Metadata["queryParameters"] = componentExt.ScalerSpec.QueryParameters
+			}
+			if componentExt.ScalerSpec.ServerAddress != "" {
+				trigger.Metadata["threshold"] = strconv.Itoa((scaleTarget))
+			}
+		}
+	} else {
+		// set trigger metadata for other triggerTypes
+		trigger.Metadata["value"] = strconv.Itoa((scaleTarget))
+		trigger.MetricType = metricType
+	}
+
+	// set queryTime for graphite trigger (if set)
+	if triggerType == "graphite" {
+		if componentExt.ScalerSpec.QueryTime != "" {
+			trigger.Metadata["queryTime"] = componentExt.ScalerSpec.QueryTime
+		}
+	}
+
+	triggers = append(triggers, trigger)
+	return triggers
+}
+
+func createKedaScaledObject(clientset kubernetes.Interface, componentMeta metav1.ObjectMeta,
+	componentExtension *v1beta1.ComponentExtensionSpec) *kedav1alpha1.ScaledObject {
+	metricConfig, err := v1beta1.NewMetricsConfig(clientset)
+	if err != nil {
+		return nil
+	}
+
+	triggers := getKedaMetrics(metricConfig, componentMeta, componentExtension)
+	annotations := componentMeta.GetAnnotations()
+
+	MinReplicas := componentExtension.MinReplicas
+	MaxReplicas := componentExtension.MaxReplicas
+	if componentExtension.ScalerSpec != nil {
+		MinReplicas = componentExtension.ScalerSpec.MinReplicas
+		MaxReplicas = componentExtension.ScalerSpec.MaxReplicas
+	}
+
+	if MinReplicas == nil {
+		MinReplicas = &constants.DefaultMinReplicas
+	}
+
+	if MaxReplicas < *MinReplicas {
+		MaxReplicas = *MinReplicas
+	}
+
+	scaledobject := &kedav1alpha1.ScaledObject{
```

Comment:
What about `.Spec.Advanced.HorizontalPodAutoscalerConfig.Behavior.*` properties etc? I suspect users would want to apply scale down rules.